src/sentry/sentry_metrics/metrics_interface.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+from typing import Mapping, Sequence, Union
+
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+
+
+class GenericMetricsBackend(ABC):
```

Comment:
Since this interface is going to produce on Kafka it will be asynchronous. Still you will likely want to allow the client to wait for the message to be acknowledged (callback). How do you plan to support this use case?

You will likely want an interface that allows you to swap the backend (relay or kafka) but still allow delivery guarantee. How will it look like in relay ? Will the call be synchronous ? 

There are multiple ways to achieve this:
- have a sync api and an async api 
- have only an async api and simulate the async behavior in relay with a background thread that wait for the HTTP response.

Reply:
I think, to start, maybe we can have an aysnc api and a sync api. I would start working on this while building the Kafka backend, which will be the next PR. Is there anything else specific to this PR that would need to change?